geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala

```diff
@@ -324,15 +324,19 @@ object QueryPlanner extends LazyLogging {
   }
 
   /**
-   * Checks for attribute transforms in the query and sets them as hints if found
-   *
-   * @param query query
-   * @param sft simple feature type
-   * @return
-   */
-  def setQueryTransforms(query: Query, sft: SimpleFeatureType) =
-    if (query.getProperties != null && !query.getProperties.isEmpty) {
-      val (transformProps, regularProps) = query.getPropertyNames.partition(_.contains('='))
+    * Checks for attribute transforms in the query and sets them as hints if found
+    *
+    * @param query query
+    * @param sft simple feature type
+    * @return
+    */
+  def setQueryTransforms(query: Query, sft: SimpleFeatureType) = {
+    import scala.collection.JavaConversions._
+    val properties = query.getPropertyNames
+    query.setProperties(Query.ALL_PROPERTIES)
+    if (properties != null && properties.nonEmpty &&
+        properties.toSeq != sft.getAttributeDescriptors.map(_.getLocalName)) {
```

Comment:
@elahrvivaz that would make sense. I think an SFT is ordered by definition since you can reference things by index.